Terminal.Gui/Core/View.cs

```diff
@@ -2894,9 +2894,14 @@ public virtual bool OnMouseEvent (MouseEvent mouseEvent)
 				return false;
 			}
 
-			var args = new MouseEventArgs (mouseEvent);
-			if (OnMouseClick (args))
-				return true;
+			if (mouseEvent.Flags.HasFlag (MouseFlags.Button1Clicked) || mouseEvent.Flags.HasFlag (MouseFlags.Button2Clicked)
```

Comment:
Much more expensive than necessary, even in Release build configuration.

Aside from performance, it is not atomic to make multiple calls on it unless you make a local copy of it first and, therefore, unless the first one returns true (thus skipping evaluation of the rest since it's `||`), it exposes a race condition.
 
Especially when testing more than one flag, the bitwise operations should be preferred, and should be combined to one check against a constant value of the same underlying type as the enum, if possible. The compiler actually doesn't treat any of those values as constants, here (really seems like it should, doesn't it???), and does a lot of work that causes a lot of casting and copying, which is from the mouseEvent object on the heap, for each call to HasFlag.

Declaring a const int or whatever backs that enum as your mask value (which can still be written with the friendly names) and then making a single `(mouseEvent.Flags & thatMask) != 0` check saves a surprising amount of work and eliminates the race condition on this line.

Alternatively, use of a switch statement for cases like this one is also pretty good and, so long as there's not a guard clause on any case explicitly using the mouseEvent.Flags value, avoids the race condition, as well.